### ISSUE TITLE
Accessibility Improvements

### ISSUE THREAD

**Issue by Alex-Bubblemaster** [telerik/kendo-ui-core#3372]
Accessibility Improvements are currently under review.
Planned enhancements will be added here upon planning completion.  
- [x] [Hierarchy grid should have area-expanded attributes](https://github.com/telerik/kendo-ui-core/issues/3369)
- [x] [Kendo UI Grid MVC does not render a guid value as an id attribute for the th element, which matches the aria-describedby value of the cells in that column](https://github.com/telerik/kendo-ui-core/issues/3475)
- [x] [Grouping and ungrouping grid keyboard support ](https://github.com/telerik/kendo-ui-core/issues/4479)
- [x] [Grid with details should notify the users of the Expand/Collapse functionality](https://github.com/telerik/kendo-ui-core/issues/4480)
